nservicebus/sql-persistence/oracle-caveats.md

```diff
@@ -0,0 +1,52 @@
+---
+title: Oracle Caveats
+component: SqlPersistence
+related:
+reviewed: 2017-04-06
+---
+
+The SQL Persistence provides true autonomy between endpoints by using separate tables for every endpoint based on the endpoint name. However, due to Oracle's 30-character limit on table names and index names, the SQL Persistence must make some compromises.
+
+## Table Names
+
+NOTE: For a complete example of the schema created by the SQL Persistence for Oracle, see [Oracle Scripts](oracle-scripts.md).
+
+For storing subscriptions, timeouts, and outbox data, SQL Persistence will reserve 24 characters for the endpoint name, leaving 3 characters for the persister type, and and additional 3 characters for an index type.Names are then constructed as `{EndpointName}{PersisterTypeSuffix}{KeyType}`.
+
+The following table shows table names created for an endpoint named `My.Endpoint`:
+
+| Persister     | Suffix |    Table Name    |       Indexes       |
+|---------------|:------:|:----------------:|:-------------------:|
+| Subscriptions |  `_SS` | `MY_ENDPOINT_SS` | `MY_ENDPOINT_SS_PK` |
+| Timeouts      |  `_TO` | `MY_ENDPOINT_TO` | `MY_ENDPOINT_TO_PK`<br/>`MY_ENDPOINT_TK`<br/>`MY_ENDPOINT_SK` |
+| Outbox        |  `_OD` | `MY_ENDPOINT_OD` | `MY_ENDPOINT_OD_PK`<br/>`MY_ENDPOINT_IX` |
+
+If an endpoint name is longer than 24 characters, an exception will be thrown, and a substitute [table prefix](index.md#installation-table-prefix) must be specified in the endpoint configuration:
+
+snippet: TablePrefix
+
+### Sagas
+
+Tables generated for sagas reserve 27 characters for the saga name, leaving 3 characters for the `_PK` suffix for the table's primary key.
+
+In order to accommodate as many characters for the saga name as possible, the [table prefix](index.md#installation-table-prefix) is not represented in the resulting table name.
+
+| Saga Class Name |   Table Name  |    Primary Key   |
+|-----------------|:-------------:|:----------------:|
+| OrderPolicy     | `ORDERPOLICY` | `ORDERPOLICY_PK` |
+
+A 3-character suffix is not enough to uniquely identify multiple correlation properties in a deterministic way, so unfortunately index names for sagas cannot be named after the owner table in the same way as for other persisters.
+
+Index names for correlation properties are constructed using the prefix `SAGAIDX_` plus the first 22 characters of a SHA1 hash of the saga name and correlation property name. The owning table for a particular index can be discovered by querying the database:
```

Comment:
I could see being more generic, but is it worth at least stating that if you change your saga name or your correlation property name, the index name is going to change as well?